src/main/java/org/jenkinsci/plugins/durabletask/ProcessLiveness.java

```diff
@@ -43,6 +54,23 @@
      * @return true if it is apparently still alive (or we cannot tell); false if it is believed to not be running
      */
     public static boolean isAlive(VirtualChannel channel, int pid, Launcher launcher) throws IOException, InterruptedException {
+        Boolean working = workingLaunchers.get(launcher);
+        if (working == null) {
+            // Check to see if our logic correctly reports that an unlikely PID is not running.
+            working = !_isAlive(channel, 9999, launcher);
```

Comment:
It makes sense to mark this code as a HACK in any case. Should be an applicable temporary solution, but needs a better fix since there is a corner-case